assets/ts/components/actionbar/actionbar.component.ts

```diff
@@ -0,0 +1,396 @@
+// @ts-nocheck
+import extendDialogs from "../../modules/dialogs";
+
+// Data layer Web component created
+window.dataLayer = window.dataLayer || [];
+window.dataLayer.push({
+  "event": "customElementRegistered",
+  "element": "action bar"
+});
+
+function setSelectAllInput(element, value){
+
+  if(element && value == "all"){
+            
+    element.querySelector('input').indeterminate = false;
+    element.querySelector('input').checked = true;
+    element.querySelector('label').textContent = `Select all`;
+  }
+  else if(element && value == 0){
+
+    element.querySelector('input').indeterminate = false;
+    element.querySelector('input').checked = false;
+    element.querySelector('label').textContent = `Select all`;
+  }
+  else if(element && value){
+
+    element.querySelector('input').indeterminate = true;
+    element.querySelector('input').checked = false;
+    element.querySelector('label').textContent = `${value} item${value > 1 ? 's' : ''} selected`;
+  }
+  else if(element){
+    
+    element.querySelector('input').checked = false;
+    element.querySelector('input').indeterminate = false;
+    element.querySelector('label').textContent = `Select all`;
+  }
+}
+
+class iamActionbar extends HTMLElement {
+
+  constructor(){
+    super();
+    this.attachShadow({ mode: 'open'});
+    
+    const assetLocation = document.body.hasAttribute('data-assets-location') ? document.body.getAttribute('data-assets-location') : '/assets'
+    const coreCSS = document.body.hasAttribute('data-core-css') ? document.body.getAttribute('data-core-css') : `${assetLocation}/css/core.min.css`;
+    const loadCSS = `@import "${assetLocation}/css/components/actionbar.css";`;
+
+    const template = document.createElement('template');
+    template.innerHTML = `
```

Comment:
@JamesLambert1986  Is this `innerHtml` going to cause issues again for the vue stuff? Or because you've added functions that emit specifically will this be okay?